### ISSUE TITLE
How to have part of the NLP Gradients and Jacobians Manually Calculated?

### ISSUE THREAD

**Issue by warrenhanwen** [casadi/casadi#3390]
Hi developers,
First thank you for developing such a powerful library.

I'm working on a nonlinear optimal control problem where some of the costs and constraints only have numerical gradients and jacobians. Is there any convenient way I can have the gradients and jacobians to be partly calculated by Casadi's built-in symbolic differentiation, and partly evaluated by me, while still using casadi's interface to solvers?

Thank you!

**Comment by jaeandersson** [comment]
Yes, it's possible. You can formulate the costs and/or constraints without the problematic entries and then manually create any derivative information. You would then have to yourself manually add the contribution from the entries you define yourself and then map back to CasADi Function objects. Best is if can be done symbolically, otherwise you can create a `Callback` function. Finally, you need to map the derivative information to the solver, e.g. by setting the `cache` option in your `nlpsol` call.
